File: src/main/java/com/graqr/threshr/model/redsky/stores/MailingAddress.java
package com.graqr.threshr.model.redsky.stores;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public record MailingAddress(
        @JsonProperty("address_line1")
        String addressLine1,
        String city,
        @JsonProperty("country_code")
        String countryCode,
        String region,
        String state,
        @JsonProperty("postal_code")
        String postalCode,
        @Nullable String country) {
}

File: src/main/kotlin/com/graqr/threshr/circleStore/model/store/MailingAddress.kt
package com.graqr.threshr.circleStore.model.store

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.core.annotation.Nullable
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class MailingAddress(
    @param:JsonProperty("address_line1")
    val addressLine1: String,
    val city: String,
    @param:JsonProperty("country_code")
    val countryCode: String,
    val region: String,
    val state: String,
    @param:JsonProperty("postal_code")
    val postalCode: String,
    @param:Nullable val country: String
) {
}
